From 7228e7653c0ad51897310d59646658c19dfae26f Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Wed, 11 Dec 2013 12:42:03 -0800
Subject: sign_target_files_apks: rewrite build.prop correctly in eng builds

In eng builds, ro.display.id has many space separated items and was
resulting in an error when trying to rewrite it as 'value' gets
turned into a list and never converted back to a string.

Category: AOSP bug fix
Domain: AOSP-build
Upstream-Candidate: Yes
Issue: ADEV-360
Change-Id: I6c8633ed2eb52c56a4097992a32d53d80df4f844
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
---
 tools/releasetools/sign_target_files_apks |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/tools/releasetools/sign_target_files_apks b/tools/releasetools/sign_target_files_apks
index a0aedde..cdeb3c3 100755
--- a/tools/releasetools/sign_target_files_apks
+++ b/tools/releasetools/sign_target_files_apks
@@ -466,8 +466,9 @@ def RewriteProps(data):
       elif key == "ro.build.display.id":
         # change, eg, "JWR66N dev-keys" to "JWR66N"
         value = value.split()
-        if len(value) == 2 and value[1].endswith("-keys"):
-          value = value[0]
+        if len(value) >  1 and value[-1].endswith("-keys"):
+          value.pop()
+        value = " ".join(value)
       line = key + "=" + value
     if line != original_line:
       print "  replace: ", original_line
-- 
1.7.4.1

